internal/resources/business_unit/resource.go

```diff
@@ -0,0 +1,934 @@
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/labd/commercetools-go-sdk/platform"
+	"github.com/labd/terraform-provider-commercetools/internal/utils"
+)
+
+var (
+	_ resource.Resource                = &companyResource{}
+	_ resource.ResourceWithConfigure   = &companyResource{}
+	_ resource.ResourceWithImportState = &companyResource{}
+)
+
+type companyResource struct {
+	client *platform.ByProjectKeyRequestBuilder
+}
+
+// Schema implements resource.Resource.
+func (b *companyResource) Schema(_ context.Context, req resource.SchemaRequest, res *resource.SchemaResponse) {
+	res.Schema = schema.Schema{
+		Description: "Business Unit type to represent the top level of a business. Contains specific fields and values that differentiate a Company from the generic BusinessUnit.\n\n" +
+			"See also the [Business Unit API Documentation](https://docs.commercetools.com/api/projects/business-units",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "Unique identifier of the Company.",
+				Computed:    true,
+			},
+			"version": schema.Int64Attribute{
+				Description: "The current version of the Company.",
+				Computed:    true,
+			},
+			"key": schema.StringAttribute{
+				Description: "User-defined unique identifier for the Company.",
+				Required:    true,
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the Company.",
+				Required:    true,
+			},
+			"status": schema.StringAttribute{
+				Description: "The status of the Company.",
+				Required:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf(BusinessUnitActive, BusinessUnitInactive),
+				},
+			},
+			"contact_email": schema.StringAttribute{
+				Description: "The email address of the Company.",
+				Optional:    true,
+			},
+			"shipping_address_ids": schema.ListAttribute{
+				Description: "List of the shipping addresses used by the Company.",
+				Optional:    true,
+				ElementType: types.StringType,
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"billing_address_ids": schema.ListAttribute{
+				Description: "List of the billing addresses used by the Company.",
+				Optional:    true,
+				ElementType: types.StringType,
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"default_shipping_address_id": schema.StringAttribute{
+				Description: "ID of the default shipping Address.",
+				Optional:    true,
+			},
+			"default_billing_address_id": schema.StringAttribute{
+				Description: "ID of the default billing Address.",
+				Optional:    true,
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"associate": schema.ListNestedBlock{
```

Comment:
I think this whole block is missing the `customer` block: https://docs.commercetools.com/api/projects/business-units#associatedraft